tests/api/test_api.py

```diff
@@ -0,0 +1,185 @@
+                             request_controller=MockRequestController, payload_type=MockDto, job_method=PickableMock())
+    job_service.add_to_queue = Mock(
+        return_value=EricaAuftragDto(type=request_type, status=Status.new, payload="{}",
+                                     request_id=request_id))
+    with patch(get_job_service_patch(endpoint), MagicMock(return_value=job_service)):
+        response = await method(data)
+        assert response == location + str(request_id)
+
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("method, request_type", [(get_fsc_request_job, RequestType.freischalt_code_request),
+                                                  (get_fsc_activation_job, RequestType.freischalt_code_activate)])
+async def test_if_get_fsc_request_or_activation_job_returns_success_status(method, request_type):
+    request_id = uuid.uuid4()
+    idnr = "test_idnr"
+    elster_request_id = "test_elster_request_id"
+    transfer_ticket = "test_transfer_ticket"
+    erica_request = EricaRequest(type=request_type, status=Status.success,
+                                 payload={"idnr": idnr},
+                                 result={"elster_request_id": elster_request_id,
+                                         "transfer_ticket": transfer_ticket},
+                                 request_id=request_id, creator_id="test")
+    with patch("erica.api.v2.endpoints.fsc.get_erica_request", MagicMock()) as mock_get_request:
+        mock_get_request.return_value = erica_request
+        response = await method(request_id)
+        assert response.processStatus == JobState.SUCCESS
+        assert response.result.idnr == idnr
+        assert response.result.elster_request_id == elster_request_id
+        assert response.result.transfer_ticket == transfer_ticket
+        assert response.errorCode is None
+        assert response.errorMessage is None
+
+
+@pytest.mark.asyncio
+async def test_if_get_fsc_revocation_job_returns_success_status():
+    request_id = uuid.uuid4()
+    idnr = "test_idnr"
+    transfer_ticket = "test_transfer_ticket"
+    erica_request = EricaRequest(type=RequestType.freischalt_code_revocate, status=Status.success,
+                                 payload={"idnr": idnr},
+                                 result={"transfer_ticket": transfer_ticket, "idnr": idnr},
+                                 request_id=request_id, creator_id="test")
+    with patch("erica.api.v2.endpoints.fsc.get_erica_request", MagicMock()) as mock_get_request:
+        mock_get_request.return_value = erica_request
+        response = await get_fsc_revocation_job(request_id)
+        assert response.processStatus == JobState.SUCCESS
+        assert response.result.idnr == idnr
+        assert response.result.transfer_ticket == transfer_ticket
+        assert response.errorCode is None
+        assert response.errorMessage is None
+
+
+@pytest.mark.asyncio
+async def test_if_get_tax_validity_job_returns_success_status():
```

Comment:
Just another idea: do you think it could make sense to group some of these tests together in classes?
Normally, We'd group them by use case (request + activation) - this this does not really makes sense for parametrize, maybe group these to together in a class like "TestStatusCode" or something?

Reply:
I would group them into use cases, so we match the files to the endpoint files! Meaning, test_fsc, test_est, etc!

Reply:
I just realized we could group the tests into use cases but that would mean loosing the power of parameterized tests and therefore duplicating a lot of code. Just have a look into the remodeled tests and let me know what you think :)